awx/main/management/commands/host_metric.py

```diff
@@ -33,17 +174,39 @@ def handle(self, *args, **options):
         if until is not None:
             filter_kwargs['last_automation__lte'] = until
 
-        result = HostMetric.objects.filter(**filter_kwargs)
+        filter_kwargs_host_metrics_summary = {}
+        if since is not None:
+            filter_kwargs_host_metrics_summary['date__gte'] = since
+        if until is not None:
+            filter_kwargs_host_metrics_summary['date__lte'] = until
+
+        host_metric_row_count = HostMetric.objects.filter(**filter_kwargs).count()
```

Comment:
these 2 db consuming queries can be omitted if there will be loop with increasing db offset and checked emptiness of retrieved queries.

Reply:
This should be doable.